lab_4_retrieval_w_clustering/main.py

```diff
@@ -389,6 +644,10 @@ def make_report(self, num_examples: int, output_path: str) -> None:
             output_path (str): path to output file
         """
 
+        clusters_info = self.__algo.get_clusters_info(num_examples)
+        with open(output_path, "w", encoding="utf-8") as file:
+            json.dump(clusters_info, file)
```

Comment:
do you know the difference between ```dump``` and ```dumps```?

Reply:
`Dump` is used when the object needs to be stored in a file, while `dumps` is used when the object needs to be string